README.md

```diff
@@ -1,19 +1,90 @@
+  This will proxy to a running SonarQube instance, but allow you to use your own local JavaScript instead of what was bundled with your plugin. Once started, you can access `http://localhost:3000` in your browser, and use SonarQube as you normally would.  
+  You can use a different port by using the `PORT` environment variable. Example:  
+  ```
+  PORT=6060 npm start
+  ```
+  You can control to which SonarQube instance you proxy to by setting the `PROXY_URL` environment variable to any valid URL (defaults to `http://localhost:9000`). Example:  
+  ```
+  PROXY_URL=https://sonarqube.example.com npm start
+  ```
+* `npm test` to start watching your files for changes, and run tests accordingly.
+* `npm run build` to build your front-end code.  
+  Usually, you will not need to call this; instead, this should be part of your package building process.  
+  See Back-end > Building above.
+
+### Building
+
+This example plugin uses [Webpack](https://webpack.js.org/) for building the final JavaScript. Whatever build system you choose to use, the final result *MUST* adhere to the following rules:
+
+* 1 entry file *per extension page*.
+* The name of each entry file must correspond to the `page_id` of the registered page (see `src/main/java/org/sonarsource/plugins/example/web/MyPluginPageDefinition.java` and compare with the entry points in `conf/webpack/webpack.config.js`).
+* Each entry file must be located in the resulting JAR's `static/` folder.
+
+The building process should be included in your full packaging process. In this example plugin, `mvn package` will call `npm run build` prior to finalizing the JAR package.
+
+### Testing
+
+This project uses [Jest](https://jestjs.io/) for testing. Running `npm test` will run Jest in `--watch` mode. You can find the configuration for Jest in `package.json`.
+
+### How to use these files
+
+It is recommended you check out the sources in `src/main/js/` directly. The code is well commented, and provides real-world examples on how to interact with SonarQube.
+
+The pages are registered in `src/main/java/org/sonarsource/plugins/example/web/MyPluginPageDefinition.java`, and their respective front-end source code is located in `src/main/js/`. These examples use different stacks to demonstrate different possibilities:
+
+* React JS examples (recommended, SonarQube uses React v16.8):
+  * `src/main/js/portfolio_page/`
+  * `src/main/js/admin_page/`
+* Backbone JS example: `src/main/js/project_page/`
+* Vanilla JS example: `src/main/js/global_page/`
+
+#### Helper APIs exposed by SonarQube
+
+There are several helper APIs exposed by SonarQube, like functions to make authenticated API requests, or parsing measures data. SonarQube also exposes several React components, in case you want to use React for your own pages.
+
+You can find the full list of exposed helpers [here](https://github.com/SonarSource/sonarqube/blob/master/server/sonar-web/src/main/js/app/components/extensions/exposeLibraries.ts).
```

Comment:
Gloups... I really don't think we want to advertise that too much, I think we should drop most of them if possible in favor of sonar-ui-common (when it will be ready).

Advertising those add contraints on all those exposed components to be backward compatible...

I think the only one that is currently documented and that we have to support today is SonarRequest (Jay had to create a legacy copy of it to be able to apply changes that we wanted for us).

Reply:
See below.